saleor/graphql/checkout/mutations/checkout_delivery_method_update.py

```diff
@@ -104,6 +104,11 @@ def perform_on_shipping_method(
             checkout_info, lines, shipping_method=delivery_method, collection_point=None
         )
 
+        # Clear checkout shipping address if it was C&C before.
+        if checkout.collection_point_id:
+            checkout.shipping_address = None
+            checkout.save(update_fields=["shipping_address"])
+
```

Comment:
Seems like we can move it at the beginning of `_update_delivery_method` to avoid the additional call to DB for saving the checkout.